src/slskd/Integrations/Scripts/ScriptService.cs

```diff
@@ -64,45 +80,40 @@ private async Task HandleEvent(Event data)
             _ = Task.Run(() =>
             {
                 Process process = default;
+                var processId = Guid.NewGuid();
 
                 try
                 {
                     var run = script.Value.Run;
+                    string executable = run.Executable ?? DefaultExecutable;
+                    string args = run.Command;
 
-                    // split the string into at most 2 parts, leaving part [0] the executable and part [1] the rest of the string, but possibly null
-                    var parts = run.Split(" ", count: 2, options: StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.RemoveEmptyEntries);
-
-                    var executable = parts[0];
-                    string args = default;
-
-                    if (parts.Length > 1)
+                    if (string.IsNullOrEmpty(executable))
                     {
-                        args = parts[1].Replace("$DATA", data.ToJson());
+                        Log.Warning("Script '{Script}' will not be run: unable to determine script executable. Update the script configuration, or set your operating system's SHELL envar.");
+                        return;
                     }
 
-                    Log.Debug("Running script '{Script}': {Run}", script.Key, run);
+                    Log.Debug("Running script '{Script}': \"{Executable}\" {Args} (id: {ProcessId})", script.Key, executable, args, processId);
                     var sw = Stopwatch.StartNew();
 
                     process = new Process()
                     {
-                        StartInfo = new ProcessStartInfo(fileName: executable)
+                        StartInfo = new ProcessStartInfo(executable ?? DefaultExecutable)
                         {
+                            WorkingDirectory = Program.ScriptDirectory,
                             UseShellExecute = false,
+                            CreateNoWindow = true,
                             RedirectStandardOutput = true,
                             RedirectStandardError = true,
-                            RedirectStandardInput = true,
+                            Arguments = args,
```

Comment:
I know nothing about csharp (so  feel free to ignore this comment), but isn't [ArgumentList](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.argumentlist?view=net-9.0#system-diagnostics-processstartinfo-argumentlist) just more practical than `Arguments`?

Reply:
it would be but unfortunately there's only `Arguments` which is a string

Reply:
What about that? https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.-ctor?view=net-9.0#system-diagnostics-processstartinfo-ctor(system-string-system-collections-generic-ienumerable((system-string))) Again, sorry if I'm talking gibberish: I really know nothing about the .net environment